### PR TITLE
Make TOC Visible

### PR DESCRIPTION
**Pull request by NPDebs**
This PR proposes a fix for issue #415
The table of content across all pages on the website did not display any content. This should fix that.

Screenshot (339)

Screenshot (335)

Screenshot (334)

**Note for reviewers:**
The toggle button does not hide the list items.

### CODE REVIEW

**Comment by NPDebs** [comment]
Hi, @AndreaSanchezTapia & @bebatut. 👋 
This issue has been dormant for 17 days. So, I took the initiative and drafted this PR, attempting to fix it. Could you please check if everything is in order? 

Thank you so much.

**Comment by bebatut** [comment]
Sure you can take this issue if you want :smile:

**Comment by NPDebs** [comment]
> Sure you can take this issue if you want 😄

Sorry, sorry. I know I should have asked to assigned before opening the PR. 🙈
I'll go ask formally now. 🏃

**Comment by NPDebs** [comment]
@bebatut 👀🙂

**Comment by bebatut** [comment]
It does not really solve the issue as the toggle button does not hide the list items.
I think the solution is probably more complex: having the TOC on the side, directly in the template for long pages

File: _includes/toc.html [hunk 1]
```diff
@@ -6,7 +6,7 @@ <h1 class="no_toc">
     </a>
 </h1>
 <div class="content is-hidden">
-1. TOC
-{:toc}
 </div>
-</div>
\ No newline at end of file
+</div>
+* auto-gen TOC:
```
Comment (NPDebs) [inline _includes/toc.html:11]: @bebatut Can we go ahead and delete this file, as it is no longer needed to display the TOC?

Comment (bebatut) [reply _includes/toc.html:11]: Yes

File: ols-6.md [hunk 2]
```diff
@@ -119,7 +119,7 @@ Mentors advise and inspire
 ### Pool of mentors
 
 <!-- Pool of possible mentors (hidden when cohort has started)
-Any modification of the content should be done in the _data/ols-6-metadata.yaml file
+Any modification of the content should be done in the _data/ols-6-metadata.yaml file -->
```
Comment (bebatut) [inline ols-6.md:122]: Can you revert your change here?

File: ols-4.md [hunk 3]
```diff
@@ -160,7 +160,7 @@ We thank the **{{ metadata.possible-mentors | size }} persons who registered to
         {% endfor %}
     </ul>
 </div>
-{% endif %} -->
+{% endif %}
```
Comment (bebatut) [inline ols-4.md:163]: Can you revert your change here?

File: _includes/toc.html (hunk 1 above)
Comment (NPDebs) [reply _includes/toc.html:11]: Thanks.

File: ols-8.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ---
-layout: page
+layout: cohort
```
Comment (bebatut) [inline ols-8.md:2]: You probably need to remove the `include` there too

File: _config.yml [hunk 5]
```diff
@@ -43,4 +43,22 @@ includes_dir: .
 
 plugins:
   - jekyll-last-modified-at
-  - jekyll-redirect-from
\ No newline at end of file
+  - jekyll-redirect-from
+  - jekyll-toc
```
Comment (bebatut) [inline _config.yml:47]: Can you check if this plugin is allowed in GitHub pages?

File: _layouts/cohort.html [hunk 6]
```diff
@@ -0,0 +1,14 @@
+---
```
Comment (bebatut) [inline _layouts/cohort.html:1]: Could you rename the layout `page-with-toc` or something like this? So we could reuse it not only for cohorts. Thanks

Comment (NPDebs) [reply _layouts/cohort.html:1]: > Is there a way to reduce the space between the list elements in the TOC? And maybe also removing the bullets in the list

Yeah. For sure...

I think the bullets make it easier to read though. Don't you think so?

File: _config.yml (hunk 5 above)
Comment (NPDebs) [reply _config.yml:47]: Yeah, it is. I checked before using it...

File: _layouts/cohort.html (hunk 6 above)
Comment (NPDebs) [reply _layouts/cohort.html:1]: Good point. 💯

**Comment by NPDebs** [comment]
> Is there a way to reduce the space between the list elements in the TOC? And maybe also removing the bullets in the list

Yeah. For sure...

I think the bullets make it easier to read though. Don't you think so?

**Comment by bebatut** [comment]
I do not think the bullets are needed in a TOC where we expect a list. But the hierarchy should stay  definitely visible

**Comment by NPDebs** [comment]
I removed the bullets from the elements at the top of the hierarchy. It looks okay...

**Comment by bebatut** [comment]
image

Could you also remove the bullets in lower levels? Something that looks like this would be nice

image

File: Gemfile [hunk 7]
```diff
@@ -6,4 +6,5 @@ gem 'jekyll-archives', group: :jekyll_plugins
 gem 'nokogiri', group: :jekyll_plugins
 gem 'html-proofer', group: :jekyll_plugins
 gem 'jekyll-redirect-from', group: :jekyll_plugins
-gem 'webrick', group: :jekyll_plugins
\ No newline at end of file
+gem 'webrick', group: :jekyll_plugins
+gem 'jekyll-toc', group: :jekyll_plugins
```
Comment (bebatut) [inline Gemfile:10]: This is not a supported plugin for GitHub page: https://pages.github.com/versions/

**Comment by bebatut** [comment]
The linting is fixed now

**Comment by bebatut** [comment]
@NPDebs could you look at implementing for the ToC a similar ToC as in the library page? Thanks

**Comment by NPDebs** [comment]
> @NPDebs could you look at implementing for the ToC a similar ToC as in the library page? Thanks

That's exactly what I was trying to do before I ran into this Linting error. It doesn't really make sense to be having an error in this blogpost.

**Comment by bebatut** [comment]
I also do not understand why there are so many files modified in this PR.
It might be good to start the work from scratch

**Comment by bebatut** [comment]
I think it is okay, specially because most commits were about fixing stuff ;)

**Comment by bebatut** [comment]
Thanks a lot!

**Comment by bebatut** [comment]
I will close this PR in favor in #682

**Pull request closed without merging**